src/apscheduler/datastores/asyncpg.py

```diff
@@ -0,0 +1,582 @@
+        async with self.pool.acquire() as conn:
+            result = await conn.fetch(query, record_class=MyRecord)
+            tasks = [
+                Task.unmarshal(self.serializer, {k: v for k, v in row.items()})
+                for row in result
+            ]
+            return tasks
+
+    async def _deserialize_schedules(self, result: list[MyRecord]) -> list[Schedule]:
+        schedules: list[Schedule] = []
+        for row in result:
+            try:
+                schedules.append(
+                    Schedule.unmarshal(self.serializer, {k: v for k, v in row.items()})
+                )
+            except SerializationError as exc:
+                await self._event_broker.publish(
+                    ScheduleDeserializationFailed(schedule_id=row["id"], exception=exc)
+                )
+
+        return schedules
+
+    async def _deserialize_jobs(self, result: list[MyRecord]) -> list[Job]:
+        jobs: list[Job] = []
+        for row in result:
+            try:
+                jobs.append(
+                    Job.unmarshal(self.serializer, {k: v for k, v in row.items()})
+                )
+            except SerializationError as exc:
+                await self._event_broker.publish(
+                    JobDeserializationFailed(job_id=row["id"], exception=exc)
+                )
+
+        return jobs
+
+    async def get_schedules(self, ids: set[str] | None = None) -> list[Schedule]:
+        async for attempt in self._retry():
+            with attempt:
+                async with self.pool.acquire() as conn:
+                    async with conn.transaction():
+                        if ids:
+                            query = """select * from schedules where id=any($1) order by id"""  # noqa: E501
+                            result = await conn.fetch(query, ids, record_class=MyRecord)
+                        else:
+                            query = """select * from schedules order by id"""
+                            result = await conn.fetch(query, record_class=MyRecord)
+                        return await self._deserialize_schedules(result)
+
+    async def add_schedule(
+        self, schedule: Schedule, conflict_policy: ConflictPolicy
+    ) -> None:
+        event: DataStoreEvent
+        values = schedule.marshal(self.serializer)
+        query = """insert into schedules(id, task_id, trigger,
+        args, kwargs, coalesce,
+        misfire_grace_time, max_jitter,
+        tags, next_fire_time, last_fire_time,
+        acquired_by, acquired_until)
+        values ($1, $2, $3, $4, $5, $6, $7, $8, $9, $10, $11, $12, $13)
+        on conflict (id) do update set task_id=$2, trigger=$3,
+        args=$4, kwargs=$5, coalesce=$6,
+        misfire_grace_time=$7, max_jitter=$8,
+        tags=$9, next_fire_time=$10, last_fire_time=$11,
+        acquired_by=$12, acquired_until=$13
+        where schedules.id=$1
+        returning case when xmax::text::int > 0 then 'updated' else 'inserted' end
```

Comment:
same cool stuff with on conflict do update and returning which case we are in :)